code/api/src/main/scala/com/avast/datadog4s/api/event/Event.scala

```diff
@@ -0,0 +1,79 @@
+package com.avast.datadog4s.api.event
+
+import java.time.Instant
+
+import com.avast.datadog4s.api.event
+
+/**
+ *
+ * @param title required - The title of the event
+ * @param text required - The text body of the event
+ * @param date required (defaults to the current time from the DogStatsD server) - The epoch timestamp for the event
+ * @param hostname The name of the host
+ * @param aggregationKey A key to use for aggregating events
+ * @param priority Specifies the priority of the event (normal or low).
+ * @param sourceTypeName The source type name
+ * @param alertType error, warning, success, or info (defaults to info)
+ */
+sealed trait Event {
```

Comment:
This is a way more complicates than it needs to be. Why not only a simple case class:
```
final case class Event(
  title: String
  text: String
  date: Option[Instant]          = None
  hostname: Option[String]       = None
  aggregationKey: Option[String] = None
  priority: event.Priority.Value = Priority.Normal
  sourceTypeName: Option[String] = None
  alertType: AlertType.Value     = AlertType.Info
}
```

Reply:
i think the proposed implementation is more user friendly ... it's almost like a logging library. Yes the code is a little more complex but the usage is simpler

Reply:
I disagree. If you want simple usage, you can provide helper methods inside companion object.

Reply:
I could live with helpers on companion object